doc/guides/maintaining-the-build-files.md

```diff
@@ -0,0 +1,386 @@
+# Maintaining the Build files
+
+This document explains how to maintain the build files in the codebase.
+
+## Overview
+
+On how to build the Node.js core, see [Building Node.js](../../BUILDING.md).
+
+There are three main build files that will be directly run to build Node.js:
+
+On Windows:
+
+- vcbuild.bat: A Windows Batch Script that locates build tools, provides
+  a subset of the targets avilable in the [Makefile](#makefile), and a few
+  targets of its own. This script also calls [configure](#configure) before
+  building the Node.js binary. For a detailed guide on this script,
+  see [vcbuild.bat](#vcbuild.bat).
+
+On unix-like systems:
+
+- configure: A (mostly) Python 2 script that accepts a set of options,
+  configure and prepare the build using gyp. For a detailed guide on
+  this script, see [configure](#configure).
+- Makefile: A makefile that can be run with BSD make or GNU make. It
+  provides a set of targets that build and test the Node.js
+  binary, produce releases and documentation, and interact with the
+  CI to run benchamarks or tests. For a detailed guide on this file,
+  see [Makefile](#makefile)
+
+## vcbuild.bat
+
+To see the help text, run `.\vcbuild help`. Consider updating this file
+when you need to update the build and testing process on Windows.
+
+```
```

Comment:
We should document the similarity/differences between  `vcbuild.bat` and `Makefile`, though not necessarily in this PR (I am not familiar enough with `vcbuild.bat` to be comfortable to write a guide on it anyway)

Reply:
that'd be quite a job for someone! one of it's biggest problems is that it lacks most of the smarts that make brings to the table and we only do a partial job of emulating the bits that are needed